docker/docker-compose.yml

```diff
@@ -153,19 +162,50 @@ services:
     environment:
       - http.max_content_length=10mb
   s3:
-    image: fingershock/fakes3:0.2.4
+    image: minio/minio:RELEASE.2020-03-19T21-49-00Z
     volumes:
-      - "s3:/fakes3_data:rw"
+      - "s3:/data:rw"
     ports:
-      - "8000"
+      - "9000"
     networks:
       - proxy
       - default
+    environment:
+      MINIO_DOMAIN: s3.localhost,altis.dev,s3
+      MINIO_REGION_NAME: us-east-1
+      MINIO_ACCESS_KEY: admin
+      MINIO_SECRET_KEY: password
+    command: server /data
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
+      interval: 30s
+      timeout: 20s
+      retries: 3
     labels:
-      - "traefik.port=8000"
+      - "traefik.port=9000"
       - "traefik.protocol=http"
       - "traefik.docker.network=proxy"
-      - "traefik.frontend.rule=HostRegexp:s3-${COMPOSE_PROJECT_NAME:-default}.altis.dev;AddPrefix:/s3-${COMPOSE_PROJECT_NAME:-default}"
+      - "traefik.frontend.rule=HostRegexp:s3-${COMPOSE_PROJECT_NAME:-default}.altis.dev"
+  s3-make-bucket:
+    <<: *mc
+    depends_on:
+      - s3
+    command: ["mb", "local/s3-${COMPOSE_PROJECT_NAME:-default}"]
+  s3-set-policy:
+    <<: *mc
+    depends_on:
+      - s3-make-bucket
+    command: ["policy", "set", "public", "local/s3-${COMPOSE_PROJECT_NAME:-default}"]
+  s3-sync-up:
+    <<: *mc
+    depends_on:
+      - s3-make-bucket
+    command: ["mirror", "--watch", "--overwrite", "local/s3-${COMPOSE_PROJECT_NAME:-default}", "/content"]
+  s3-sync-down:
```

Comment:
Do these all need to be top level services? It doesn't seem like the docker-compose file should be used for this maybe?

Reply:
I did wonder the same but it works really well at least. The other option to have a wrapper function that does `docker run minio/mc ...` and call that from php however for the long running mirror command with `—watch` it’s essentially a service.

Reply:
Found a way to get the best of all worlds. The main benefit to the previous appraoch was using docker-compose `depends_on` to ensure the bucket was created and the policy set however it's possible to use `entrypoint` instead and execute a string of commands. Removes the need for any messing around in PHP still.